src/emu/save.h

```diff
@@ -267,7 +267,11 @@ class rewinder
 // template specializations to enumerate the fundamental atomic types you are allowed to save
 ALLOW_SAVE_TYPE_AND_ARRAY(char)
 ALLOW_SAVE_TYPE          (bool); // std::vector<bool> may be packed internally
+#ifdef __sun
+ALLOW_SAVE_TYPE_AND_ARRAY(signed char)
+#else
```

Comment:
I don't understand this one. Why does the variable type "s8" work ok but not this? Also, AFAIK it is an int8_t, not "signed char". Would ALLOW_SAVE_TYPE_AND_ARRAY(osd::s8) work?

Reply:
@happppp it seems that Solaris 2.x define `char` as a distinct type from both `[un]signed char`. Weird.

Reply:
I'll also take a look at the `s/signed char/osd::s8/` thing too

Reply:
Already changed to osd::* a few weeks ago =)

Reply:
`char` is *always* distinct from both `signed char` and `unsigned char`.  Whether it's signed or not depends on the architecture (signed on x86, unsigned on PowerPC for example).  `uint8_t` and `int8_t` will be aliases for two of these, but which ones depends on your C runtime library, OS and architecture.